interfaces/builtin/content.go

```diff
@@ -169,23 +168,15 @@ func (iface *contentInterface) path(attrs interfaces.Attrer, name string) []stri
 // $SNAP_COMMON. If there are no variables then $SNAP is implicitly assumed
 // (this is the behavior that was used before the variables were supporter).
 func resolveSpecialVariable(path string, snapInfo *snap.Info) string {
-	if strings.HasPrefix(path, "$SNAP/") || path == "$SNAP" {
-		// NOTE: We use dirs.CoreSnapMountDir here as the path used will be always
-		// inside the mount namespace snap-confine creates and there we will
-		// always have a /snap directory available regardless if the system
-		// we're running on supports this or not.
-		// TODO parallel-install: use of proper instance/store name
-		return strings.Replace(path, "$SNAP", filepath.Join(dirs.CoreSnapMountDir, snapInfo.InstanceName(), snapInfo.Revision.String()), 1)
+	// Content cannot be mounted at arbitrary locations, validate the path
+	// for extra safety.
+	if err := snap.ValidatePathVariables(path); err == nil && strings.HasPrefix(path, "$") {
+		// Starts with $SNAP, $SNAP_DATA, $SNAP_COMMON, no other $VARs
```

Comment:
Perhaps adjust to be something like:
```
// ValidatePathVariables() ensures path starts with $SNAP, $SNAP_DATA,
// $SNAP_COMMON, no other $VARs and ExpandSnapVariables() only
// expands $SNAP, $SNAP_DATA and $SNAP_COMMON
```

Feel free to break this comment up as desired. The point is, I had to read ValidatePathVariables to remind myself what it did and why it is ok to only check for `strings.HasPrefix(path, '$')`, so a clarified comment might be handy for future maintenance.